common/src/nostr/tests.rs

```diff
@@ -0,0 +1,110 @@
+
+/// Receive a message when only one mesage was sent by the escrow.
+#[tokio::test]
+async fn receive_1_message() -> anyhow::Result<()> {
+    let mut buyer_nostr_client = create_nostr_client(BUYER_NSEC).await;
+    let escrow_nostr_client = create_nostr_client(ESCROW_NSEC).await;
+
+    let msg1 = TestMessage1("message 1".to_string());
+    escrow_nostr_client
+        .client
+        .send_private_msg(
+            buyer_nostr_client.public_key(),
+            serde_json::to_string(&msg1)?,
+            None,
+        )
+        .await?;
+
+    let result: TestMessage1 = buyer_nostr_client.receive_escrow_message(10).await?;
+    assert_eq!(result, msg1);
+    Ok(())
+}
+
+/// Send 2 messages and then receive them in reverse order, forcing to use the cache.
+#[tokio::test]
+async fn receive_2_messages_from_cache() -> anyhow::Result<()> {
+    let mut buyer_nostr_client = create_nostr_client(BUYER_NSEC).await;
+    let escrow_nostr_client = create_nostr_client(ESCROW_NSEC).await;
+
+    tokio::time::sleep(Duration::from_millis(50)).await; //needed to wait till the local test relay sets up both subxcriptions
+
+    let msg1 = TestMessage1("message 1".to_string());
+    escrow_nostr_client
+        .client
+        .send_private_msg(
+            buyer_nostr_client.public_key(),
+            serde_json::to_string(&msg1)?,
+            None,
+        )
+        .await?;
+
+    let msg2 = TestMessage2(1);
+    escrow_nostr_client
+        .client
+        .send_private_msg(
+            buyer_nostr_client.public_key(),
+            serde_json::to_string(&msg2)?,
+            None,
+        )
+        .await?;
+
+    assert_eq!(buyer_nostr_client.messages_cache.len(), 0);
+    let result_msg2: TestMessage2 = buyer_nostr_client.receive_escrow_message(10).await?;
+    assert_eq!(result_msg2, msg2);
+    assert_eq!(
+        buyer_nostr_client.messages_cache.len(),
+        1,
+        "After receiving for the second sent message, must be 1 message left in the cache!"
+    );
+
+    let result_msg1: TestMessage1 = buyer_nostr_client.receive_escrow_message(10).await?;
+    assert_eq!(result_msg1, msg1);
+    assert_eq!(
+        buyer_nostr_client.messages_cache.len(),
+        0,
+        "After receiving for the first sent message, must be no messages left in the cache!"
+    );
+    Ok(())
+}
+
+#[tokio::test]
+async fn spam_protection() -> anyhow::Result<()> {
+    let mut buyer_nostr_client = create_nostr_client(BUYER_NSEC).await;
+    let escrow_nostr_client = create_nostr_client(ESCROW_NSEC).await;
+
+    for i in 0..CACHE_SIZE + 1 {
+        escrow_nostr_client
+            .client
+            .send_private_msg(
+                buyer_nostr_client.public_key(),
+                serde_json::to_string(&TestMessage2(i))?,
+                None,
+            )
+            .await?;
+    }
+
+    let msg_res = buyer_nostr_client
+        .receive_escrow_message::<TestMessage1>(10)
+        .await;
+    assert!(msg_res.is_err()); //Timeout
```

Comment:
This assert fails for me because:
```rust
[common/src/nostr/tests.rs:96:5] &msg_res = Ok(
    TestMessage1(
        "message 1",
    ),
)
```
May it receives messages from the previous tests because they all use the same nsec?

Reply:
This is strange, are you running my fork of bucket and also using the `run-test.sh` script? The original from Coracle.social doesn't support the limit filter we use and the tests must also run sequentially or as single test. `cargo test`runs the tests in parallel by default. Otherwise the test can get an old or concurrent message from the relay. Using my fork and the test script the tests runs reliably for me.

Reply:
Yes i tested with your fork and the script.

Reply:
I tried it again with different nsecs for spam_protection and now it works, so the reason for the test failing is definitely using the same nsec as the previous tests

Reply:
Ok, then there is still a bug in the bucket relay :( I don't get it. But you bring me to a better approach. If we use random nsecs for each test, they can even run in parallel. Should I rewrite them this way?

Reply:
I pushed this change, please test again when possible. It think it is a better approach with the random nsec.

Reply:
If it is ok for you, I would remove the test script and update the readme.

Reply:
Makes sense to me, makes testing even easier (and faster)